src/main/java/com/cloudbees/jenkins/plugins/awscredentials/AmazonWebServicesCredentialsBinding.java

```diff
@@ -131,39 +131,40 @@ protected Class<AmazonWebServicesCredentials> type() {
     @Override
     public MultiEnvironment bind(@NonNull Run<?, ?> build, FilePath workspace, Launcher launcher, TaskListener listener)
             throws IOException, InterruptedException {
-        AWSCredentialsProvider provider = getCredentials(build);
+        AwsCredentialsProvider provider = getCredentials(build);
         if (!StringUtils.isEmpty(this.roleArn)) {
             provider = this.assumeRoleProvider(provider);
         }
 
-        AWSCredentials credentials = provider.getCredentials();
+        AwsCredentials credentials = provider.resolveCredentials();
 
         Map<String, String> m = new HashMap<String, String>();
-        m.put(accessKeyVariable, credentials.getAWSAccessKeyId());
-        m.put(secretKeyVariable, credentials.getAWSSecretKey());
+        m.put(accessKeyVariable, credentials.accessKeyId());
+        m.put(secretKeyVariable, credentials.secretAccessKey());
 
         // If role has been assumed, STS requires AWS_SESSION_TOKEN variable set too.
-        if (credentials instanceof AWSSessionCredentials) {
-            m.put(SESSION_TOKEN_VARIABLE_NAME, ((AWSSessionCredentials) credentials).getSessionToken());
+        if (credentials instanceof AwsSessionCredentials) {
+            m.put(SESSION_TOKEN_VARIABLE_NAME, ((AwsSessionCredentials) credentials).sessionToken());
         }
         return new MultiEnvironment(m);
     }
 
-    private AWSSessionCredentialsProvider assumeRoleProvider(AWSCredentialsProvider baseProvider) {
-        AWSSecurityTokenService stsClient = AWSCredentialsImpl.buildStsClient(baseProvider);
+    private AwsCredentialsProvider assumeRoleProvider(AwsCredentialsProvider baseProvider) {
```

Comment:
The automated migration tool failed here and I am just guessing at the correct APIs. Should be manually tested before and after this PR to confirm there are no regressions.

Reply:
I took a closer look at this and I think the API is straightforward enough to proceed.